rust/src/combinators/composition/sequential/ffi.rs

```diff
@@ -78,3 +114,54 @@ pub extern "C" fn opendp_combinators__make_sequential_composition(
 
     make_sequential_composition(input_domain, input_metric, output_measure, d_in, d_mids).into()
 }
+
+
+impl<QI: 'static + TotalOrd + Clone + Send + Sync, QO: 'static + TotalOrd + Clone + Send + Sync>
+    Measurement<AnyDomain, Queryable<Measurement<AnyDomain, AnyObject, TypedMetric<QI>, TypedMeasure<QO>>, AnyObject>, AnyMetric, AnyMeasure>
+{
+    pub fn into_any_queryable_map(self) -> Measurement<AnyDomain, Queryable<AnyMeasurement, AnyObject>, AnyMetric, AnyMeasure> {
+        let function = self.function.clone();
+
+        Measurement::new(
+            self.input_domain.clone(),
+            Function::new_fallible(
+                move |arg: &AnyObject| -> Fallible<Queryable<AnyMeasurement, AnyObject>> {
+                    let mut inner_qbl = function.eval(arg)?;
+
+                    Queryable::new(move |_self, query: Query<AnyMeasurement>| match query {
+                        Query::External(query) => {
+                            let privacy_map = query.privacy_map.clone();
+                            let meas = Measurement::new(
+                                query.input_domain.clone(),
+                                query.function.clone(),
+                                TypedMetric::<QI>::new(query.input_metric.clone())?,
+                                TypedMeasure::<QO>::new(query.output_measure.clone())?,
+                                PrivacyMap::new_fallible(move |d_in: &QI| {
+                                    privacy_map.eval(&AnyObject::new(d_in.clone()))?.downcast()
+                                })
+                            )?;
+                            inner_qbl
+                                .eval(&meas)
+                                .map(Answer::External)
+                        },
+                        Query::Internal(query) => {
+                            let Answer::Internal(a) =
+                                inner_qbl.eval_query(Query::Internal(query))?
+                            else {
+                                return fallible!(
+                                    FailedFunction,
+                                    "internal query returned external answer"
+                                );
+                            };
+                            Ok(Answer::Internal(a))
+                        }
+                    })
+                },
+            ),
+            self.input_metric.clone(),
+            self.output_measure.clone(),
+            self.privacy_map.clone(),
+        )
+        .expect("AnyDomain is not checked for compatibility")
```

Comment:
Surprising that this message is specific to AnyDomain, when we don't know here if that's always what would cause this to panic.